source/Public/Get-SqlDscServiceType.ps1

```diff
@@ -0,0 +1,54 @@
+<#
+    .SYNOPSIS
+        Returns all the registered service types.
+
+    .DESCRIPTION
+        Returns all the registered service types.
+
+    .OUTPUTS
+        `[System.Object[]]`
+
+    .EXAMPLE
+        Get-SqlDscServiceType
+
+        Returns all service types.
+#>
+function Get-SqlDscServiceType
+{
+    [CmdletBinding()]
+    [OutputType([System.Object[]])]
+    param ()
+
+    $registeredServiceTypes = @()
+
+    $registeredServiceType = Get-ChildItem -Path 'HKLM:\SOFTWARE\Microsoft\Microsoft SQL Server\Services'
```

Comment:
If the path does not exist it should not fail with error `Get-ChildItem : Cannot find path 'HKEY_LOCAL_MACHINE\SOFTWARE\Microsoft\Microsoft SQL Server\Services' because it does not exist. 
`. Could output a warning or just return `$null`.